app/contracts/relations/base_contract.rb

```diff
@@ -65,7 +63,7 @@ def validate_to_exists
 
     def validate_nodes_relatable
       if (model.from_id_changed? || model.to_id_changed?) &&
-         WorkPackage.relatable(model.from, model.relation_type).where(id: model.to).empty?
+         WorkPackage.relatable(model.from, model.relation_type, ignored_relation: model).where(id: model.to).empty?
```

Comment:
Calling `model.to` will force loading the `to` relation resulting in an extra query. Using `model.to_id` will not result in an extra query.